### PR TITLE
build: Build in Docker with operator-sdk v0.18.1

### PR DESCRIPTION
**Pull request by JoelColledge**
Version v0.18 of operator-sdk introduced a new structure that allows
builds to run entirely in Docker without a go build environment on the
host. This is useful for CI, so switch to using it. The new structure
seems to be undocumented, but it uses kubebuilder. These changes are
based on a basic kubebuilder project.

---

We can probably simplify the GitHub CI parts now too. I'll look into it.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# Copy the go source
+COPY cmd/ cmd/
+COPY pkg/ pkg/
+COPY version/ version/
+
+# Build
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o manager ./cmd/manager/main.go
+
+
+FROM debian:buster
```
[the remark this answers is not in the document]
Comment (JoelColledge) [reply Dockerfile:21]: All our piraeus images are based on `debian:buster`. I imagine a much simpler image would work here, but I don't want to change that with this restructuring.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+# and so that source changes don't invalidate our downloaded layer
+RUN go mod download
+
+# Copy the go source
+COPY cmd/ cmd/
+COPY pkg/ pkg/
+COPY version/ version/
+
+# Build
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o manager ./cmd/manager/main.go
```
[the remark this answers is not in the document]
Comment (JoelColledge) [reply Dockerfile:18]: This line just comes from the kubebuilder scaffold. I can't see why they should be necessary.

File: Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (WanzenBug) [reply Dockerfile:21]: :+1:

**Comment by JoelColledge** [comment]
> the operator-sdk in our CI pipeline is still v0.16.0

Unfortunately that isn't the end of the story. If we use operator-sdk v0.18.1 with this branch, there is no `generate crds` command. It would seem that they are in the middle of a significant change to the way the CLI works and it is enabled when the `PROJECT` file is present. I find it totally bizarre that the available commands would depend on pwd where I happen to run the binary.

I think I might abandon this approach and just change linstor-operator-builder to build in Docker without using `operator-sdk build` at all.

**Pull request closed without merging**